### PR TITLE
Refactor reactive limits and Injection connectivity infos

### PR DESCRIPTION
**Pull request by YenguiSeddik**

### CODE REVIEW

File: src/main/java/org/gridsuite/modification/server/dto/BatteryCreationInfos.java [hunk 1]
```diff
@@ -68,15 +67,6 @@ public class BatteryCreationInfos extends InjectionCreationInfos {
     @Schema(description = "Reactive capability curve")
     private Boolean reactiveCapabilityCurve;
 
-    @Schema(description = "Connection Name")
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/dto/BatteryCreationInfos.java:71]: This move of connectionName, connectionDirection and connectionPosition fields up to InjectionCreationInfos class does not really concern the reactive limits refactoring ...
Same remark for the same move in the other dto classes and for the same move in the entity classes ...

File: src/main/java/org/gridsuite/modification/server/dto/ReactiveLimitsHolderInfos.java [hunk 2]
```diff
@@ -0,0 +1,13 @@
+package org.gridsuite.modification.server.dto;
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/dto/ReactiveLimitsHolderInfos.java:1]: Missing license and header

File: src/main/java/org/gridsuite/modification/server/modifications/BatteryCreation.java [hunk 3]
```diff
@@ -163,86 +136,11 @@ private void addExtensionsToBattery(BatteryCreationInfos batteryCreationInfos, B
         }
         reportBatteryConnectivity(batteryCreationInfos, subReporter);
         Reporter subReporterLimits = reportBatteryActiveLimits(batteryCreationInfos, subReporter);
-        createBatteryReactiveLimits(batteryCreationInfos, battery, subReporterLimits);
+        createReactiveLimits(batteryCreationInfos, battery, subReporterLimits);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/modifications/BatteryCreation.java:139]: ModificationUtils.getInstance().createReactiveLimits ? and no static keyword in this method call ?

File: src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java [hunk 4]
```diff
@@ -174,7 +141,7 @@ private void addExtensionsToGenerator(GeneratorCreationInfos generatorCreationIn
         }
         reportGeneratorConnectivity(generatorCreationInfos, subReporter);
         Reporter subReporterLimits = reportGeneratorActiveLimits(generatorCreationInfos, subReporter);
-        createGeneratorReactiveLimits(generatorCreationInfos, generator, subReporterLimits);
+        createReactiveLimits(generatorCreationInfos, generator, subReporterLimits);
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/modifications/GeneratorCreation.java:144]: Same remark as above

File: src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java [hunk 5]
```diff
@@ -814,5 +814,137 @@ public void checkMaxReactivePowerGreaterThanMinReactivePower(MinMaxReactiveLimit
             throw new NetworkModificationException(exceptionType, errorMessage + "maximum reactive power " + maxReactivePower + " is expected to be greater than or equal to minimum reactive power " + minReactivePower);
         }
     }
+
+    private static NetworkModificationException makeEquipmentException(NetworkModificationException.Type errorType,
```
Comment (FranckLecuyer) [inline src/main/java/org/gridsuite/modification/server/modifications/ModificationUtils.java:818]: remove static keyword and use singleton instance, to be consistent with the other method calls
same remark for the following methods ...

**Review by FranckLecuyer — CHANGES_REQUESTED** [review]

**Review by FranckLecuyer — APPROVED** [review]

**Review by Walid-Sahnoun — APPROVED** [review]
code ok
test ok

**Pull request merged**